pymc_marketing/clv/models/beta_geo_beta_binom.py

```diff
@@ -0,0 +1,364 @@
+
+
+# TODO: Proofread docstrings
+class BetaGeoBetaBinomModel(CLVModel):
+    r"""Beta-Geometric Beta Binomial Distribution (BG/BB) model for discrete, non-contractual transactions.
+
+    In the BG/BB model, the frequency of customer purchases is modelled as the time
+    of each purchase has an instantaneous probability of occurrence (hazard) and, at
+    every purchase, a probability of "dropout", i.e. no longer being a customer.
+
+    Customer-specific data needed for statistical inference include 1) the total
+    number of purchases (:math:`x`) and 2) the time of the last, i.e. xth, purchase. The
+    omission of purchase times :math:`t_1, ..., t_x` is due to a telescoping sum in the
+    exponential function of the joint likelihood; see Section 4.1 of [1] for more
+    details.
+
+    Parameters
+    ----------
+    customer_id: array_like
+        Customer labels. Must not repeat.
+    frequency: array_like
+        The number of purchases of customers.
+    recency: array_like
+        The time of the last, i.e. xth, purchase.
+    T: int
+        The number of purchase opportunities.
+    purchase_hyperprior: scalar PyMC distribution, optional
+        PyMC prior distribution, created via `.dist()` API. Defaults to
+        `pm.HalfFlat.dist()`
+    churn_hyperprior: scalar PyMC distribution, optional
+        PyMC prior distribution, created via `.dist()` API. Defaults to
+        `pm.HalfFlat.dist()`
+
+    Examples
+    --------
+    BG/BB model for customer
+
+    .. code-block:: python
+        import pymc as pm
+        from pymc_marketing.clv import BetaGeoModel
+
+        model = BetaGeoBetaBinom(
+            frequency=[4, 0, 6, 3, ...],
+            recency=[30.73, 1.72, 0., 0., ...]
+            p_prior=pm.HalfNormal.dist(10),
+            q_prior=pm.HalfNormal.dist(10),
+            v_prior=pm.HalfNormal.dist(10),
+        )
+
+        model.fit()
+        print(model.fit_summary())
+
+        # Estimating the expected number of purchases for a randomly chosen
+        # individual in a future time period of length t
+        expected_num_purchases = model.expected_num_purchases(
+            t=[2, 5, 7, 10],
+        )
+
+        # Predicting the customer-specific number of purchases for a future
+        # time interval of length t given their previous frequency and recency
+        expected_num_purchases_new_customer = model.expected_num_purchases_new_customer(
+            t=[5, 5, 5, 5],
+            frequency=[5, 2, 1, 8],
+            recency=[7, 4, 2.5, 11],
+            T=[10, 8, 10, 22],
+        )
+
+    References
+    ----------
+    .. [1] Fader, Peter S., Bruce G.S. Hardie, and Jen Shang (2010),
+       "Customer-Base Analysis in a Discrete-Time Noncontractual Setting,"
+       Marketing Science, 29 (6), 1086-1108. https://www.brucehardie.com/papers/020/fader_et_al_mksc_10.pdf
+    """
+
+    _model_name = "BG/BB"  # Beta-Geometric Beta-Binomial Model
+    _params = ["alpha", "beta", "gamma", "delta"]
```

Comment:
This addition is related to #128 IIRC. Is that correct? I believe that the existing models don't have this `_params` structure but it could be a nice way of closing #128 with additional methods to the base `CLVModel` class. Was what you had in mind?

Reply:
https://github.com/pymc-labs/pymc-marketing/issues/182 you mean? I wanted to get draft PRs for this and https://github.com/pymc-labs/pymc-marketing/pull/177 up before opening a PR for that one. After considering https://github.com/pymc-labs/pymc-marketing/pull/197, some major test restructurings could be underway, so I think I'll start work on https://github.com/pymc-labs/pymc-marketing/issues/182 next week.

In the meantime, I could use some help with `logp` in this PR.

Reply:
Oops yes #182, sorry. And sounds good.

I see that the `logp` utilizes `scan`, which I'm not super familiar with. I see that @ricardoV94 is giving some good pointers below